src/ParallelAlgorithms/DiscontinuousGalerkin/InitializeMortars.hpp

```diff
@@ -0,0 +1,179 @@
+            Requires<db::tag_is_retrievable_v<::Tags::InitialExtents<dim>,
+                                              DataBox>> = nullptr>
+  static auto apply(DataBox& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using simple_tags =
+        db::AddSimpleTags<mortar_data_tag,
+                          ::Tags::Mortars<::Tags::Next<temporal_id_tag>, dim>,
+                          ::Tags::Mortars<::Tags::Mesh<dim - 1>, dim>,
+                          ::Tags::Mortars<::Tags::MortarSize<dim - 1>, dim>>;
+    using compute_tags = db::AddComputeTags<>;
+
+    const auto& element = db::get<::Tags::Element<dim>>(box);
+    const auto& temporal_id = get<temporal_id_tag>(box);
+    const auto& initial_extents = db::get<::Tags::InitialExtents<dim>>(box);
+
+    db::item_type<mortar_data_tag> mortar_data{};
+    db::item_type<::Tags::Mortars<::Tags::Next<temporal_id_tag>, dim>>
+        mortar_next_temporal_ids{};
+    db::item_type<::Tags::Mortars<::Tags::Mesh<dim - 1>, dim>> mortar_meshes{};
+    db::item_type<::Tags::Mortars<::Tags::MortarSize<dim - 1>, dim>>
+        mortar_sizes{};
+    const auto& interface_meshes =
+        db::get<::Tags::Interface<::Tags::InternalDirections<dim>,
+                                  ::Tags::Mesh<dim - 1>>>(box);
+    for (const auto& direction_and_neighbors : element.neighbors()) {
+      const auto& direction = direction_and_neighbors.first;
+      const auto& neighbors = direction_and_neighbors.second;
+      for (const auto& neighbor : neighbors) {
+        const auto mortar_id = std::make_pair(direction, neighbor);
+        mortar_data[mortar_id];  // Default initialize data
+        mortar_next_temporal_ids.insert({mortar_id, temporal_id});
+        mortar_meshes.emplace(
+            mortar_id,
+            dg::mortar_mesh(
+                interface_meshes.at(direction),
+                domain::Initialization::create_initial_mesh(
+                    initial_extents, neighbor, neighbors.orientation())
+                    .slice_away(direction.dimension())));
+        mortar_sizes.emplace(
+            mortar_id,
+            dg::mortar_size(element.id(), neighbor, direction.dimension(),
+                            neighbors.orientation()));
+      }
+    }
+
+    if (AddFluxBoundaryConditionMortars) {
+      const auto& boundary_meshes =
+          db::get<::Tags::Interface<::Tags::BoundaryDirectionsInterior<dim>,
+                                    ::Tags::Mesh<dim - 1>>>(box);
+      for (const auto& direction : element.external_boundaries()) {
+        const auto mortar_id =
+            std::make_pair(direction, ElementId<dim>::external_boundary_id());
+        mortar_data[mortar_id];  // Default initialize data
```

Comment:
Was this not already default-initialized on line 123, outside the `if` statement?

Reply:
this initializes the mortar data on _external_ boundaries. Line 123 does the same on boundaries between elements.